File: src/main/java/ro/fasttrackit/homework/PersonValidator.java
package ro.fasttrackit.homework;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class PersonValidator {
    public void verify(String firstName, String lastName, int age, String city) {
        verifyNotNull(firstName);
        verifyNotNull(lastName);
        verifyNotNull(city);
        verifyName(firstName);
        verifyName(lastName);
        verifyAge(age);
        verifyName(city);
    }

    private void verifyName(String element) {
        verifyNotNull(element);
        Pattern pattern = Pattern.compile("^[ A-za-z'\\-]+$");
        Matcher matcher = pattern.matcher(element);
        if (!matcher.matches()) {
            throw new IllegalArgumentException(element + " is not an accepted argument");
        }
    }

    private void verifyAge(int age) {
        if (age < 0 || age > 120) {
            throw new IllegalArgumentException("Invalid age");
        }
    }

    private void verifyNotNull(String element) {
        if (element == null || element.equals("")) {
            throw new IllegalArgumentException("Invalid empty argument");
        }
    }

}

File: src/main/java/ro/fasttrackit/kotlinhomework/PersonValidator.kt
package ro.fasttrackit.kotlinhomework

import java.util.regex.Matcher
import java.util.regex.Pattern

class PersonValidator {
    fun verify(firstName: String, lastName: String, age: Int, city: String) {
        verifyNotNull(firstName)
        verifyNotNull(lastName)
        verifyNotNull(city)
        verifyName(firstName)
        verifyName(lastName)
        verifyAge(age)
        verifyName(city)
    }

    private fun verifyName(element: String) {
        verifyNotNull(element)
        val pattern: Pattern = Pattern.compile("^[ A-za-z'\\-]+$")
        val matcher: Matcher = pattern.matcher(element)
        if (!matcher.matches()) throw IllegalArgumentException("$element + is not an accepted argument")
    }

    private fun verifyAge(age: Int) {
        if (age < 0 || age > 120) throw IllegalArgumentException("Invalid age")
    }

    private fun verifyNotNull(element: String?) {
        if (element == "" || element == null) throw IllegalArgumentException("Invalid empty argument")
    }
}